### PR TITLE
pool domain should always return a domain

### PR DESCRIPTION
**Pull request by baohx2000**
Refs https://github.com/spulec/moto/issues/3706

### CODE REVIEW

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @baohx2000, thanks for the PR!
Would you mind adding a test to verify that this behaves as it should?

If I'm understanding the test case correctly, you can just add a single assertion to `tests/test_cognitoidp/test_cognitoidp.py::test_describe_user_pool_domain`

File: tests/test_cognitoidp/test_cognitoidp.py [hunk 1]
```diff
@@ -137,6 +137,7 @@ def test_create_user_pool_domain():
     user_pool_id = conn.create_user_pool(PoolName=str(uuid.uuid4()))["UserPool"]["Id"]
     result = conn.create_user_pool_domain(UserPoolId=user_pool_id, Domain=domain)
     result["ResponseMetadata"]["HTTPStatusCode"].should.equal(200)
+    result["CloudFrontDomain"].should_not.be.none
```
Comment (bblommers) [inline tests/test_cognitoidp/test_cognitoidp.py:140]: Typo - this should be `result["DomainDescription"]["CloudFrontDomain"]`

**Review by bblommers — CHANGES_REQUESTED** [review]

File: tests/test_cognitoidp/test_cognitoidp.py (hunk 1 above)
Comment (baohx2000) [reply tests/test_cognitoidp/test_cognitoidp.py:140]: Kind of confused... that structure would not match the next test result
`result["CloudFrontDomain"].should.equal("e2c343b3293ee505.cloudfront.net")`

The test log is gigantic and takes forever to load on github. Will try to take a look again when I get to the office.

Comment (baohx2000) [reply tests/test_cognitoidp/test_cognitoidp.py:140]: Fixed

**Review by bblommers — APPROVED** [review]
LGTM

**Pull request merged**